Content.Shared/Ninja/Systems/SharedNinjaGlovesSystem.cs

```diff
@@ -0,0 +1,181 @@
+using Content.Shared.Actions;
+using Content.Shared.CombatMode;
+using Content.Shared.Doors.Components;
+using Content.Shared.DoAfter;
+using Content.Shared.Examine;
+using Content.Shared.Hands.Components;
+using Content.Shared.IdentityManagement;
+using Content.Shared.Interaction;
+using Content.Shared.Interaction.Components;
+using Content.Shared.Interaction.Events;
+using Content.Shared.Inventory.Events;
+using Content.Shared.Ninja.Components;
+using Content.Shared.Popups;
+using Content.Shared.Research.Components;
+using Content.Shared.Timing;
+using Content.Shared.Toggleable;
+using Robust.Shared.Timing;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Content.Shared.Ninja.Systems;
+
+public abstract class SharedNinjaGlovesSystem : EntitySystem
+{
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] protected readonly SharedAppearanceSystem Appearance = default!;
+    [Dependency] private readonly SharedCombatModeSystem _combatMode = default!;
+    [Dependency] protected readonly SharedDoAfterSystem _doAfter = default!;
+    [Dependency] protected readonly SharedInteractionSystem Interaction = default!;
+    [Dependency] private readonly SharedSpaceNinjaSystem _ninja = default!;
+    [Dependency] protected readonly SharedPopupSystem Popup = default!;
+    [Dependency] private readonly UseDelaySystem _useDelay = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<NinjaGlovesComponent, GetItemActionsEvent>(OnGetItemActions);
+        SubscribeLocalEvent<NinjaGlovesComponent, ExaminedEvent>(OnExamined);
+        SubscribeLocalEvent<NinjaGlovesComponent, GotUnequippedEvent>(OnUnequipped);
+
+        // TODO: maybe move into r&d server???
+        SubscribeLocalEvent<NinjaDownloadComponent, InteractionAttemptEvent>(OnDownload);
+    }
+
+    /// <summary>
+    /// Disable glove abilities and show the popup if they were enabled previously.
+    /// </summary>
+    public void DisableGloves(EntityUid uid, NinjaGlovesComponent comp)
+    {
+        if (comp.User != null)
+        {
+            var user = comp.User.Value;
+            comp.User = null;
```

Comment:
Don't need to cache user you can just get the containing slot from inventory system from memory.

Reply:
isnt that the other way around, for getting slot from the inventory + slot name